File: medium/0050_pow/solution.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        if x == 0:
            return 0
        if n == 0:
            return 1
        if n == 1:
            return x

        result = self.myPow(x, abs(n) // 2)
        result *= result

        if n % 2 == 1:
            result *= x

        if n < 0:
            result = 1 / result

        return result

File: medium/0050_pow/solution.cpp
using namespace std;

#include <stdlib.h>

class Solution {
public:
  double myPow(double x, int n) {
    if (x == 0)
      return 0;
    if (n == 0)
      return 1;
    if (n == 1)
      return x;

    long N = abs((long)n);

    double result = myPow(x, N / 2);
    result *= result;

    if (N % 2 == 1)
      result *= x;

    if (n < 0)
      result = 1 / result;

    return result;
  }
};
